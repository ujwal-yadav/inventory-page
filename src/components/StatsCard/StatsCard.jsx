import { useTheme } from '@emotion/react';
import { Paper, Typography, useMediaQuery } from '@mui/material';
import React from 'react';

const StatsCard = ({ title, value, color }) => {
  const theme = useTheme();
  const isLargeView = useMediaQuery(theme.breakpoints.up('md'));

  return (
    <Paper
      sx={{
        padding: isLargeView ? 3 : 2,
        flex: 1,
        borderRadius: 2,
      }}
    >
      <Typography color={color} marginBottom={1} variant="h6">
        {title}
      </Typography>
      <Typography variant="h5" component={'h2'} fontWeight={'700'}>
        {value}
      </Typography>
    </Paper>
  );
};

export default StatsCard;
