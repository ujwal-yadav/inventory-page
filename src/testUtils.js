import { createTheme, ThemeProvider } from '@mui/material';
import { render } from '@testing-library/react';

export const renderWithTheme = (ui, { themeOptions = {} } = {}) => {
  const theme = createTheme(themeOptions);
  return render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);
};
