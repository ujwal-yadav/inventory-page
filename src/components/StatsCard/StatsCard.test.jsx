import React from 'react';
import StatsCard from './StatsCard';
import '@testing-library/jest-dom/extend-expect';
import { renderWithTheme } from '../../testUtils';

describe('StatsCard', () => {
  it('should render correctly', () => {
    const { asFragment } = renderWithTheme(
      <StatsCard title="Orders" value="200" color="primary" />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
