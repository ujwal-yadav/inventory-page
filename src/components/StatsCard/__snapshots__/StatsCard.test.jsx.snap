// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsCard should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-bzp9q3-MuiPaper-root"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 css-asfi9w-MuiTypography-root"
    >
      Orders
    </h6>
    <h2
      class="MuiTypography-root MuiTypography-h5 css-bezmlo-MuiTypography-root"
    >
      200
    </h2>
  </div>
</DocumentFragment>
`;
